com.unity.shadergraph/Editor/Data/Graphs/ShaderDropdown.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor.ShaderGraph.Internal;
+
+namespace UnityEditor.ShaderGraph
+{
+    [Serializable]
+    class ShaderDropdown : ShaderInput
+    {
+        public ShaderDropdown()
+        {
+            this.displayName = "Dropdown";
+            // Add sensible default entries for Enum type
+            m_Entries = new List<DropdownEntry>();
+            m_Entries.Add(new DropdownEntry(1, "A"));
+            m_Entries.Add(new DropdownEntry(2, "B"));
+        }
+
+        [SerializeField]
+        private List<DropdownEntry> m_Entries;
+
+        public List<DropdownEntry> entries
+        {
+            get => m_Entries;
+            set => m_Entries = value;
+        }
+
+        public override bool allowedInMainGraph { get => false; }
+
+        [SerializeField]
+        private int m_Value;
+
+        public int value
+        {
+            get => m_Value;
+            set => m_Value = value;
```

Comment:
I noticed the reordering thing, same thing happens for keyword enums, so I just decided to leave it as-as, since it's super easy to fix the default right there when you're changing the ordering. The one thing I **did** specifically address is reordering items affecting the selected dropdown in parent graphs. I don't store by index like enums do, so dependent graphs should not be affected by a simple reordering of the items, like they do for materials.

Regarding m_Value being out of range, I'll have another look today and see if that's a scenario that could occur when indexing into the list, and fix it as needed.

Reply:
Where do you think this might need clamping? I'm not sure I see a failure point for this.

Reply:
By the way, I added a fix for reordering thing, so prevent the default value changing.

Reply:
```
public string entryName
{
    get => entries[value].displayName;
}
```
uses value unguarded so if someone ever sets value to something invalid it will break. I'm not saying this does happen currently, but this is an api that allows things to break.

Reply:
Alright, I'll put a clamp in there to be sure.